### ISSUE TITLE
Some question about FPT in RenderTrans

### ISSUE THREAD

**Issue by wangjue-wzq** [ZHANGDONG-NJUST/FPT#25]
In rendering_trans.py, the forward method input X_high and X_low in RenderTrans with different size, but after self.str_conv3x3(X_low) operation, the stride=2(line 40-44 in rendering_trans.py). But in line 86-90 of FPT.py, such as self.rt_p5_p3, self.rt_p5_p2, The feature size of P5, P3 and P2 are 1x, 4x and 8x, while self.str_conv3x3 only performs double scaling reduction. Therefore, the error of inconsistent feature scales appears in rendering_trans.py lin 44. How to solve it? Thank you！

**Comment by visionxyz** [comment]
Hi, have you solved this problem? Thanks!

**Comment by wangjue-wzq** [comment]
> Hi, have you solved this problem? Thanks!

No, the code has a lot of errors.

**Comment by RokiaAbdeen** [comment]
same issue

**Comment by ZHANGDONG-NJUST** [comment]
For cross-layer (more than 2 layers) feature interaction, it is necessary to use stride conv several times to achieve the consistent scale.

**Comment by lassestahnke** [comment]
Hi, I encountered the same issue. I fixed it by added an additional parameter called n_down, which is the number of downsampling operations. I also combined the strided convolutions and the batch norm in a torch.Sequential container.

`        # repeat stride convolution and batch norm n_down times to account for larger size difference in feature maps
        if n_down > 0:
            self.str_conv = nn.Sequential(nn.Conv2d(channels_low,
                                                    channels_high,
                                                    kernel_size=3,
                                                    stride=2,
                                                    padding=1,
                                                    bias=False),
                                        nn.BatchNorm2d(channels_high),
                                        *[nn.Conv2d(channels_high,
                                                    channels_high,
                                                    kernel_size=3,
                                                    stride=2,
                                                    padding=1,
                                                    bias=False),
                                        nn.BatchNorm2d(channels_high)]*(n_down-1)
                                         )
        else:
            self.str_conv = nn.Sequential(
                nn.Conv2d(channels_low,
                          channels_high,
                          kernel_size=3,
                          stride=1,
                          padding=1,
                          bias=False),
                nn.BatchNorm2d(channels_high)
            )`
